wavefront/wavefront.go

```diff
@@ -51,20 +55,70 @@ type (
 		listener            net.Listener
 		server              *grpc.Server
 		reporterInitialized bool
+		reporter            wf.WavefrontMetricsReporter
 	}
 )
 
 // ensure that WavefrontAdapter implements the HandleMetricServiceServer interface.
 var _ metric.HandleMetricServiceServer = &WavefrontAdapter{}
 
 // createWavefrontReporter creates a reporter that periodically flushes metrics to Wavefront.
-func createWavefrontReporter(cfg *config.Params) {
+func (wa *WavefrontAdapter) createWavefrontReporter(cfg *config.Params) {
 	hostTags := map[string]string{"source": cfg.Source}
+
 	if direct := cfg.GetDirect(); direct != nil {
-		go wf.WavefrontDirect(metrics.DefaultRegistry, cfg.FlushInterval, hostTags, cfg.Prefix, direct.Server, direct.Token)
+		directCfg := &senders.DirectConfiguration{
+			Server:               direct.Server,
+			Token:                direct.Token,
+			FlushIntervalSeconds: int(cfg.FlushInterval),
+		}
+		sender, err := senders.NewDirectSender(directCfg)
+		if err != nil {
+			panic(err)
+		}
+
+		wa.reporter = wf.NewReporter(
+			sender,
+			application.New("istio", "istio"),
```

Comment:
Include hostTags (from line 67) when you create a reporter.  Host Tags should be sent along with all metrics reported by this reporter.
Also why is application.New("istio", "istio") needed ?

Reply:
The mentioned variable, hostTags map has only the "Source" currently.  
In the writeMetrics call, the tags are sent along with the metrics in the GetOrRegisterMetric() call.  I believe you are referring to this. Please correct me if my understanding is incorrect.

Regarding, the hardcoded value "istio" for application and service, 

The application Tag is mandatory in the call to NewReporter() call and tags (app and service names) are not in the config.Params. 

Is there any other option to address this issue?

Reply:
Discussed this with Vikram, for now we can have both application and service as "wavefront-istio-adapter".  In future we can remove them - https://github.com/wavefrontHQ/go-metrics-wavefront/issues/22

Reply:
Regarding "hostTags",  In v0.9.0, they were a way to specify tags at the reporter level and specifying the source tag i guess, meaning any tag specified will be added to all the points flushed by this reporter. This seems to have changed with the latest "go-metrics" library.

With the latest version, i think you need to specify reporter level tags using a customMap inside application tags - 
https://github.com/wavefrontHQ/wavefront-sdk-go/blob/master/application/application.go#L11 
and source can be specified using - wf.Source(<source>)

In this case we don't have any reporter level tags, and you already specify the source using "wf.Source(cfg.Source)" . So basically, now, hostTags is being used only to "createSystemStatsReporter(<hostTags>)" at line 123. . Could you move the declaration closer to its usage.

Reply:
comments incorporated